DnD-Montreal/session-tome
# Fix GM role assignment in Campaign updates

## Description
Closes #499 

Fixes the inability to switch from a character role to a GM role in a campaign (and vice versa).

## How to Test 
You can mostly follow the steps in #499, but...
1. Open a campaign in which you have a character assigned
2. update the campaign details to change the character you have assigned to "none"
3. Observe that you no longer have a character in the campaign
4. Add a new character back, observe that it is the correct character.
